lib/connections/service.go

```diff
@@ -185,18 +186,18 @@ func NewService(cfg config.Wrapper, myID protocol.DeviceID, mdl Model, tlsCfg *t
 	// the common handling regardless of whether the connection was
 	// incoming or outgoing.
 
-	service.Add(util.AsService(service.connect))
-	service.Add(util.AsService(service.handle))
+	service.Add(util.AsService(service.connect, service.String()))
```

Comment:
So in a lot of places where we create "sub"-services I think just passing on the name of the parent service isn't appropriate. These should be `fmt.Sprintf("%s/connector", service)` and so on to differentiate or the name will be meaningless?